### ISSUE TITLE
why nss and cc are negative number?

### ISSUE THREAD

**Issue by chsy1996** [atsiami/STAViS#2]
i run train_test splits on my computer, but it shows negative numbers about cc and nss, is that any problem happen?

**Comment by chsy1996** [comment]
Is the test running correctly, or is there something wrong with my operation?

**Comment by pkoutras** [comment]
In the train/test losses we take the negative values of CC, NSS since we need to minimize the total loss. In the evaluation code we compute the proper CC and NSS values.

**Comment by chsy1996** [comment]
I understand, thank you very much! !

**Comment by Sultanax** [comment]
Hi! I still do not understand. When running visual only, I am able to get positive NSS and CC metrics (running `compute_all_databases.sh` with my experiments path. However, when running audiovisual, I get negative NSS and CC. I don't understand where this is coming from. Can you help me out? @pkoutras

**Comment by pkoutras** [comment]
Do you get negative CC and NSS with audiovisual running the `compute_all_databases.sh`?
During the training the CC and NSS are negative since we need to minimize the total loss. 
After getting the audiovisual predictions (running `run_all_splits_audiovisual_test.sh`),  you need to run again the `compute_all_databases.sh`  with the new predictions path (e.g. `./experiments/audiovisual_test`).

**Comment by Sultanax** [comment]
Sorry, let me rephrase. I am running the testing only option. I added in the `--no_val` and `--no_train` options.  
These are my results when I run visual only vs audiovisual using the pre-trained paths:

I am running on Coutrot DB1 only, with just split 1. 

| Metric         | Visual       | AV          |
|---------------|-------------|-------------|
| CC           | 0.4032       | -0.3775     |
| Similarity   | 0.3624       | 0.1176      |
| NSS          | 1.6876       | -1.4458     |
| AUC_Judd     | 0.8453       | 0.2006      |
| AUC_shuffled | 0.5663       | 0.4372      |

I can see that with the audiovisual experiment results, the predicted saliency maps (jpg files) have a white background and black salient clusters, hence the negative correlation. However, the visual experiment results have a black background with white salient clusters, as it should be. Could you explain what I could be doing wrong? I don't think I changed anything about the code.

**Comment by pkoutras** [comment]
Do you use the provided scripsts? What are your commands? 

Note that there are some differences in the commands for visual vs. audiovisual testing (see attached diff file).

[visual_vs_audiovisual.txt](https://github.com/user-attachments/files/18711319/visual_vs_audiovisual.txt)

**Comment by Sultanax** [comment]
Ah I see. With audiovisual, the option should be `--resume_path` rather than `--pretrain_path`. Let me try again and I will get back to you. Thank you for the help.

Just to be sure, the `--resume_path` is essentially for transfer learning. Let's say we want to add on an additional modality to the model: we can `--resume_path` to the audiovisual pretrained path and train the different modality on top of that. Correct?

**Comment by pkoutras** [comment]
The `--pretrain_path`  loads the weights for the visual-only model, while `--audio_pretrain_path` loads the weights for the pretrained SoundNet encoder. The `--resume_path` loads the weights for the final combined audio-visual model.
